### PR TITLE
[functional_test] Add directive for testing block prologue function

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

Add support for testing block prologue transaction directly in functional test.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Replaced a few existing test cases. Will remove the assertions in the genesis once the movelang modules are ready.

### CODE REVIEW

File: language/functional-tests/src/evaluator.rs [hunk 1]
```diff
@@ -41,6 +42,13 @@ pub struct Transaction<'a> {
     pub input: String,
 }
 
+/// Commands that can be evaluated by the testing infra.
+#[derive(Debug)]
+pub enum Command<'a> {
```
[the remark this answers is not in the document]
Comment (runtian-zhou) [reply language/functional-tests/src/evaluator.rs]: Yes! The main reason in my mind is that we will add more directives over there. The most pressing ones are reloading the on-chain configurations and loading writesets. Some of them will be represented as a libra_type `Transaction` but some of them will not. The main thing here is that we want to mimic the public api of LibraVM as much as possible so there will be things not representable by `Transaction`. That's why I call it a `Command` as it sounds more like one "Directive" to me.

**Comment by runtian-zhou** [comment]
> Overall I feel like this PR is adding quite a bit of complexities to the parsing of block metadata transactions.
> 
> Right now there are only two pieces of information we need to know about the block metadata transaction, and they can fit into a one liner easily:
> 
> ```
> //! block-prologue: vivian, 1000000
> ```
> 
> Personally I would prefer this simpler model for now, assuming the block metadata won't get significantly more complex in the near future. (@runtian-zhou correct me if I am wrong.)
> 
> For a future refactor of the e2e framework, there are two things we can do to make the system more robust and flexible:
> 
> 1. Have a set of Rust APIs to configure the genesis.
> 2. Transition to a more generalized attribute syntax & object model for the global and transaction configs. Imaging writing JSON or YAML instead of the pseudo-command like syntax. With this we won't be required to add a new syntax every time we add some new functionalities.
> 
> To summarize the point I'm trying to make: we'll likely rewrite & simplify the whole parsing part soon, so try not to add more complexity than you need there if possible.

I'm not sure how we are going to do 1 in functional test IMO. We do need to invoke rust apis in functional test text. I would love to see 2 coming soon. The main reason why this PR is a bit pressing is that currently the LibraSystem just exposed the block prologue api as public just to do the testing. I would hope to add this functionality in ASAP just to be able to add the assertions back as they should.

File: language/functional-tests/src/evaluator.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (vgao1996) [reply language/functional-tests/src/evaluator.rs]: I buy that.

**Pull request closed without merging**